portal-impl/src/com/liferay/portal/db/index/IndexUpdaterUtil.java

```diff
@@ -201,6 +207,28 @@ private static void _awaitFuturesTermination() {
 		_futures.clear();
 	}
 
+	private static void _deleteDuplicateEntries(
+			String tableName, String indexesSQL)
+		throws InvalidSyntaxException {
+
+		BundleContext bundleContext = SystemBundleUtil.getBundleContext();
+
+		DuplicateRemover duplicateRemover = new PortalDuplicateRemover();
+
+		String filter = "(service.tables=portal)";
```

Comment:
The filter will be applied here.

In my testing,  I had three active components and was able to select a specific one and when ran, for the layout component, would create a layout as well as print out the layout plid.

Reply:
I must be applied inside the _db.process_ consumer code since it internally iterate over all partitions when DB Partition is enabled.